src/lib/props.js

```diff
@@ -0,0 +1,113 @@
+import {log} from './utils';
+import assert from 'assert';
+
+/**
+ * Performs equality by iterating through keys on an object and returning false
+ * when any key has values which are not strictly equal between the arguments.
+ * @param {Object} opt.oldProps - object with old key/value pairs
+ * @param {Object} opt.newProps - object with new key/value pairs
+ * @param {Object} opt.ignoreProps={} - object, keys that should not be compared
+ * @returns {null|String} - null when values of all keys are strictly equal.
+ *   if unequal, returns a string explaining what changed.
+ */
+/* eslint-disable max-statements, complexity */
+export function compareProps({oldProps, newProps, ignoreProps = {}} = {}) {
+  assert(oldProps !== undefined && newProps !== undefined, 'compareProps args');
+
+  // props are immutable, so shallow equality => deep equality
+  if (oldProps === newProps) {
+    return null;
+  }
+
+  if (typeof newProps !== 'object' || newProps === null) {
+    return 'new props is not an object';
```

Comment:
The message may be a little confusing when neither `oldProps` or `newProps` is an object.
Context: I think `diffUpdateTriggers` also uses this function.